site/src/components/MoreMenu/MoreMenu.tsx

```diff
@@ -44,23 +54,35 @@ const useMoreMenuContext = () => {
   return ctx;
 };
 
-export const MoreMenuTrigger = (props: IconButtonProps) => {
+export const MoreMenuTrigger = ({
+  children,
+  ...props
+}: HTMLProps<HTMLButtonElement>) => {
   const menu = useMoreMenuContext();
 
-  return (
-    <IconButton
-      aria-controls="more-options"
-      aria-label="More options"
-      aria-haspopup="true"
-      onClick={menu.open}
-      ref={menu.triggerRef}
-      {...props}
-    >
-      <MoreVertOutlined />
-    </IconButton>
-  );
+  return cloneElement(children as ReactElement, {
```

Comment:
Is this type-safe? My worry is that it looks like there's nothing restricting the type of `children` at the type level, so if someone accidentally passes in a string or a primitive as a prop, they might get a runtime error